dbms/src/Debug/dbgKVStore/dbgFuncInvestigator.cpp

```diff
@@ -0,0 +1,287 @@
+                ErrorCodes::BAD_ARGUMENTS);
+        }
+        size_t handle_column_size = table_info.is_common_handle ? table_info.getPrimaryIndexInfo().idx_cols.size() : 1;
+        auto key_size = arg_size / 2;
+        std::vector<Field> start_field;
+        start_field.reserve(key_size);
+        std::vector<Field> end_field;
+        end_field.reserve(key_size);
+
+        for (size_t i = 0; i < handle_column_size; i++)
+        {
+            auto & column_info = table_info.columns[table_info.getPrimaryIndexInfo().idx_cols[i].offset];
+            TiDB::DatumBumpy start_datum = TiDB::DatumBumpy(
+                RegionBench::convertField(column_info, typeid_cast<const ASTLiteral &>(*args[OFFSET + i]).value),
+                column_info.tp);
+            start_field.emplace_back(start_datum.field());
+            TiDB::DatumBumpy end_datum = TiDB::DatumBumpy(
+                RegionBench::convertField(
+                    column_info,
+                    typeid_cast<const ASTLiteral &>(*args[OFFSET + key_size + i]).value),
+                column_info.tp);
+            end_field.emplace_back(end_datum.field());
+        }
+
+        start_key = RecordKVFormat::genKey(table_info, start_field);
+        end_key = RecordKVFormat::genKey(table_info, end_field);
+    }
+    else
+    {
+        start_handle = static_cast<HandleID>(safeGet<UInt64>(typeid_cast<const ASTLiteral &>(*args[OFFSET]).value));
+        start_key = RecordKVFormat::genKey(table_id, start_handle);
+        if (args.size() == 3)
+        {
+            end_handle = start_handle + 1;
+        }
+        else
+        {
+            end_handle
+                = static_cast<HandleID>(safeGet<UInt64>(typeid_cast<const ASTLiteral &>(*args[OFFSET + 1]).value));
+        }
+        end_key = RecordKVFormat::genKey(table_id, end_handle);
+    }
+
+    auto range = RegionRangeKeys(std::move(start_key), std::move(end_key));
+    auto regions = kvstore.getRegionsByRangeOverlap(range.comparableKeys());
+
+    for (const auto & [region_id, region] : regions)
+    {
+        auto r = RegionBench::DebugRegion(region);
+        auto & data = r.debugData();
+
+        for (const auto & [k, v] : data.defaultCF().getData())
+        {
+            if (k.first == start_handle)
+            {
+                result.in_default.emplace_back(region_id, k.second);
+            }
+        }
+        for (const auto & [k, v] : data.writeCF().getData())
+        {
+            if (k.first == start_handle)
+            {
+                result.in_write.emplace_back(region_id, k.second);
+            }
+        }
+
+        auto lock_key = std::make_shared<const TiKVKey>(TiKVKey::copyFrom(start_key));
+        if (data.lockCF().getData().contains(
+                RegionLockCFDataTrait::Key{lock_key, std::string_view(lock_key->data(), lock_key->dataSize())}))
+            result.in_lock.emplace_back(region_id);
```

Comment:
```suggestion
        if (data.lockCF().getData().contains(
                RegionLockCFDataTrait::Key{lock_key, std::string_view(lock_key->data(), lock_key->dataSize())}))
        {
            result.in_lock.emplace_back(region_id);
        }
```

Reply:
Add `{...}` for it because the if statement is long and it is easy to miss change it later